packages/grafana-ui/src/components/VizLegend/VizLegendTable.tsx

```diff
@@ -23,23 +23,15 @@ export const VizLegendTable = <T extends unknown>({
   readonly,
 }: VizLegendTableProps<T>): JSX.Element => {
   const styles = useStyles2(getStyles);
+  const stats: Record<string, DisplayValue> = {};
 
-  const columns = items
-    .map((item) => {
-      if (item.getDisplayValues) {
-        return item.getDisplayValues().map((i) => i.title);
+  for (const item of items) {
+    if (item.getDisplayValues) {
+      for (const displayValue of item.getDisplayValues()) {
+        stats[displayValue.title ?? '?'] = displayValue;
```

Comment:
If you want to show both Last, and Last not null in the legend this will fail, as both calcs have the same title.

What if the [getDisplayValues in VizLegendItem](https://github.com/grafana/grafana/blob/a08ee057eb4cad9349f2168f01ebfeb7a042bfa0/packages/grafana-ui/src/components/VizLegend/types.ts#L40) was typed as:
```
getDisplayValues?: () => Array<DisplayValue & { reducerId: ReducerID }>;
```

Reducer ids are unique and we could use those as sort keys instead.